invoices/invoiceregistry.go

```diff
@@ -692,8 +717,28 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 	_, err = i.AddInvoice(invoice, ctx.hash)
```

Comment:
Isn't it that future AMP keysends won't have this payment-level hash anymore? (will the invoice data template in the db change with AMP?)

Reply:
Indeed, true AMP payments do not need to be indexed by payment hash anymore and can be referenced entirely by payment address. We do require that legacy keysend be indexed by set id tho, since there is no explicit payment address by which to query for.

Reply:
I rebased https://github.com/lightningnetwork/lnd/pull/4167 on top of this PR to see how it would work for interactive keysend acceptance. 

One thing I noticed is that `SettleInvoice`/`CancelInvoice` can still be used on converted-legacy-keysend invoices with hash and preimage. Shouldn't this be disabled to prevent users building applications that might break with amp invoices?